### PR TITLE
reduce startup time by checking lsp executables on FileType event

### PR DESCRIPTION
**Pull request by itchyny**
This pull request improves the startuptime by delaying executables check on FileType event.

### CODE REVIEW

File: plugin/lsp_settings.vim [hunk 1]
```diff
@@ -130,49 +130,61 @@ function! s:vimlsp_settings_get(name, key, default) abort
 endfunction
 
 function! s:vimlsp_setting() abort
+  for l:ft in keys(s:settings)
+    if has_key(g:, 'lsp_settings_whitelist') && index(g:lsp_settings_whitelist, l:ft) == -1 || empty(s:settings[l:ft])
+      continue
+    endif
+    exe 'augroup' s:load_or_suggest_group_name(l:ft)
+      au!
+      exe 'autocmd FileType' ft 'call s:vimlsp_load_or_suggest(' string(ft) ')'
```
Comment (mattn) [inline plugin/lsp_settings.vim]: `l:ft`

Comment (itchyny) [reply plugin/lsp_settings.vim]: oops

**Comment by mattn** [comment]
before
```
273.000  129.000  126.000: sourcing C:\Users\mattn\.vim\plugged\vim-lsp-settings\plugin\lsp_settings.vim
```
after
```
149.000  001.000  001.000: sourcing C:\Users\mattn\.vim\plugged\vim-lsp-settings\plugin\lsp_settings.vim
```
LGTM

**Comment by mattn** [comment]
Thank you

**Pull request merged**